backend/question-service/lambda/infra/index.ts

```diff
@@ -0,0 +1,108 @@
+import * as aws from "@pulumi/aws";
+import * as pulumi from "@pulumi/pulumi";
+
+// The IAM policy document for lambda
+const assumeRole = aws.iam.getPolicyDocument({
+  statements: [
+    {
+      effect: "Allow",
+      principals: [
+        {
+          type: "Service",
+          identifiers: ["lambda.amazonaws.com"],
+        },
+      ],
+      actions: ["sts:AssumeRole"],
+    },
+  ],
+});
+
+// Create an IAM role for the Lambda function
+const iamForLambda = new aws.iam.Role("iamForLambda", {
+  assumeRolePolicy: assumeRole.then((assumeRole) => assumeRole.json),
+});
+
+// The IAM policy document for CloudWatch
+const cloudwatchPolicyDocument = aws.iam.getPolicyDocument({
+  statements: [
+    {
+      actions: [
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents",
+        "logs:DescribeLogStreams",
+      ],
+      resources: ["arn:aws:logs:*:*:*"],
+      effect: "Allow",
+    },
+  ],
+});
+
+// Attach the CloudWatch policy to the role to allow CloudWatch to write logs
+new aws.iam.RolePolicy("rolePolicy", {
+  role: iamForLambda.id,
+  policy: cloudwatchPolicyDocument.then((cloudwatchPolicyDocument) => {
+    return cloudwatchPolicyDocument.json;
+  }),
+});
+
+// Get Connection String from Pulumi stack
+const currentEnv = pulumi.getStack(); // 'staging' or 'prod'
+const stackRef = new pulumi.StackReference(
+  `cs3219/question-service-infra/${currentEnv}`
+);
+const mongoConnectionString = stackRef.getOutput("mongoConnectionString");
```

Comment:
should add a check here for the deployment to fail if the stack reference/output doesn't exist 

easier to debug if the deployment of the lambda fails compared to when the lambda runs but the database no data cos the lambda couldn't access the env vars

Reply:
Resolved in https://github.com/CS3219-AY2324S1/ay2324s1-course-assessment-g03/commit/de1eb1590e1c47555ff72b5033e505e90f480449:
- Added try-catch blocks to handle errors when stack output does not exist